### PR TITLE
Convert `ha-climate-control` ot Lit and add tooltips to buttons

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

Conversion to Lit + ensure the icon buttons have SVG icons and tooltip + aria-label (the actual driver behind this "small" change 😆 ).

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: #3267 
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-climate-control.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+          >
+          </ha-icon-button>
+        </div>
+      </div>
+    `;
+  }
+
+  protected updated(changedProperties) {
+    if (changedProperties.has("value")) {
+      this.valueChanged();
```
Comment (zsarnett) [inline src/components/ha-climate-control.ts:58]:
```suggestion
      this._valueChanged();
```

File: src/components/ha-climate-control.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    }
+  }
+
+  private temperatureStateInFlux(inFlux) {
+    this._targetTemperature.classList.toggle("in-flux", inFlux);
+  }
+
+  private round(value) {
```
Comment (zsarnett) [inline src/components/ha-climate-control.ts:66]:
```suggestion
  private _round(value) {
```

File: src/components/ha-climate-control.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+  @property() public unit = "";
+
+  @property() public min?: number;
+
+  @property() public max?: number;
+
+  @property() public step = 1;
+
+  private lastChanged?: number;
```
Comment (zsarnett) [inline src/components/ha-climate-control.ts:24]:
```suggestion
  private _lastChanged?: number;
```

File: src/components/ha-climate-control.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+          >
+          </ha-icon-button>
+        </div>
+        <div>
+          <ha-icon-button
+            .path=${mdiChevronDown}
+            .label=${this.hass.localize(
+              "ui.components.climate-control.temperature_down"
+            )}
+            @click=${this.decrementValue}
```
Comment (zsarnett) [inline src/components/ha-climate-control.ts:48]:
```suggestion
            @click=${this._decrementValue}
```

File: src/components/ha-climate-control.ts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    return html`
+      <div id="target_temperature">${this.value} ${this.unit}</div>
+      <div class="control-buttons">
+        <div>
+          <ha-icon-button
+            .path=${mdiChevronUp}
+            .label=${this.hass.localize(
+              "ui.components.climate-control.temperature_up"
+            )}
+            @click=${this.incrementValue}
```
Comment (zsarnett) [inline src/components/ha-climate-control.ts:38]:
```suggestion
            @click=${this._incrementValue}
```

File: src/components/ha-climate-control.ts [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+      :host {
+        display: flex;
+        justify-content: space-between;
+      }
+      .in-flux#target_temperature {
+        color: var(--error-color);
+      }
+      #target_temperature {
+        align-self: center;
+        font-size: 200%;
```
Comment (zsarnett) [inline src/components/ha-climate-control.ts:120]: Can we set this to a value in Pixels instead?

File: src/components/ha-climate-control.ts [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        direction: ltr;
+      }
+      .control-buttons {
+        font-size: 200%;
```
Comment (zsarnett) [inline src/components/ha-climate-control.ts:124]: Can we set this to a value in Pixels instead?

File: src/components/ha-climate-control.ts [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    }
+  }
+
+  static get styles(): CSSResultGroup {
+    return css`
+      :host {
+        display: flex;
+        justify-content: space-between;
+      }
+      .in-flux#target_temperature {
```
Comment (zsarnett) [inline src/components/ha-climate-control.ts:115]:
```suggestion
      #target_temperature {
```

Comment (zsarnett) [inline src/components/ha-climate-control.ts:115]:
```suggestion
      #target_temperature.in-flux {
```

File: src/common/number/clamp.ts [hunk 9]
```diff
@@ -1,2 +1,9 @@
 export const clamp = (value: number, min: number, max: number) =>
   Math.min(Math.max(value, min), max);
+
+// Variant that only applies the clamping to a border if the border is defined
+export const conditionalClamp = (value: number, min?: number, max?: number) => {
+  value = min ? Math.max(value, min) : value;
+  value = max ? Math.min(value, max) : value;
```
Comment (bramkragten) [inline src/common/number/clamp.ts:7]: Dont change the input, create a new var

**Review by bramkragten — APPROVED** [review]

**Pull request merged**